.github/workflows/tests.yml

```diff
@@ -40,6 +41,28 @@ jobs:
         if: github.event_name == 'pull_request'
         id: formulae-detect
 
+      - name: Count pull requests with the `CI-long-timeout` label
+        if: github.event_name == 'pull_request'
+        id: count_long_timeout_prs
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
```

Comment:
Wonder if a future extension could be to ensure that we only apply this on PRs where the CI is already running?

Reply:
That would be nice, yes. But I'd rather leave that to someone who understands GitHub's GraphQL API better than I do. (This could very well be future-me.)